service/datasync/config.go

```diff
@@ -0,0 +1,35 @@
+package datasync
+
+import (
+	"github.com/onflow/flow-go/model/flow"
+)
+
+// DefaultConfig is the default configuration for the Streamer.
+var DefaultConfig = Config{
+	BufferSize:    32000000,
```

Comment:
is `BufferSize` measured in bytes or `ExecutionData` objects? `ExecutionData` average ~100KB, so 32M objects would be ~3.2TB

Reply:
https://github.com/onflow/flow-archive/blob/amlandeep/integrate-exec-data-sync-api/service/datasync/exec_data_sync.go#L154

It looks like the `BufferSize` refers to number of execution data objects. This number looks arbitrarily large, I'll reduce it down 400k, because it looks like there were about 4M blocks between mainnet19 -> 20, and also mainnet20 -> 21. This new number would be slightly less than 10% of a spork, but we could also reduce it further since I'm not sure if the buffer really needs to be this big

Reply:
400K would still be 400GB of memory. We should shoot for a number that allows the fetcher to be sufficiently ahead of the indexer so it always has enough work. I'm guessing that's somewhere <100. maybe 20ish would be a good place to start.